src/app/services/wallet.service.ts

```diff
@@ -59,15 +59,27 @@ export class WalletService {
     this.loadBalances();
   }
 
-  create(label: string, seed: string) {
-    seed = this.getCleanSeed(seed);
-    const wallet = {
-      label: label,
-      seed: seed,
-      addresses: [this.cipherProvider.generateAddress(this.ascii_to_hexa(seed))]
-    };
-    this.addWallet(wallet);
-    this.loadBalances();
+  create(label: string, seed: string): Promise<void> {
+    return new Promise<void>((resolve) => {
+      seed = this.getCleanSeed(seed);
+
+      const wallet = {
+        label: label,
+        seed: seed,
+        addresses: [this.cipherProvider.generateAddress(this.ascii_to_hexa(seed))]
+      };
+
+      this.all.first().subscribe((wallets: Wallet[]) => {
+        if (wallets.some((w: Wallet) => w.addresses[0].address === wallet.addresses[0].address)) {
+          throw new Error('trying to add an existing wallet!');
```

Comment:
Please use the same error message as the desktop wallet: "A wallet already exists with this seed".